server/src/main/java/org/opensearch/common/Randomness.java

```diff
@@ -126,18 +130,47 @@ public static Random get() {
     }
 
     /**
-     * Provides a secure source of randomness.
-     * <p>
-     * This acts exactly similar to {@link #get()}, but returning a new {@link SecureRandom}.
+     * Returns a {@link SecureRandom} via
+     * {@code
+     * FipsDRBG.SHA512_HMAC.fromEntropySource(new BasicEntropySourceProvider(entropySource, true)).build(null, true)
+     * }
+     * if BCFIPS is on classpath and the application is running in FIPS JVM,
+     * otherwise it returns a non-approved {@link SecureRandom}.
      */
     public static SecureRandom createSecure() {
-        if (currentMethod != null && getRandomMethod != null) {
-            // tests, so just use a seed from the non secure random
-            byte[] seed = new byte[16];
-            get().nextBytes(seed);
-            return new SecureRandom(seed);
-        } else {
-            return new SecureRandom();
+        try {
+            Class<?> registrarClass = Class.forName("org.bouncycastle.crypto.CryptoServicesRegistrar");
+            Method isApprovedOnlyMethod = registrarClass.getMethod("isInApprovedOnlyMode");
+            boolean approvedOnly = (Boolean) isApprovedOnlyMethod.invoke(null);
+
+            if (approvedOnly) {
+                Class<?> basicEntropyProviderClass = Class.forName("org.bouncycastle.crypto.util.BasicEntropySourceProvider");
+                Constructor<?> entropyConstructor = basicEntropyProviderClass.getConstructor(SecureRandom.class, boolean.class);
+                boolean isPredictionResistant = true;
```

Comment:
Under which conditions `isPredictionResistant ` is `false`? Could we drop it and use `true` instead ?
```suggestion
                boolean isPredictionResistant = true;
```

Reply:
I'm in favor to leave it declarative since it is instantly clear what this parameter is about and it is used twice - both instances should be instantiated the same way.

`new BasicEntropySourceProvider(entropySource, isPredictionResistant)`
`builder.build(null, isPredictionResistant)`